example/examples/CacheURLTiles.js

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import { StyleSheet, View, Text, Dimensions } from 'react-native';
+
+import MapView, {
+  MAP_TYPES,
+  PROVIDER_DEFAULT,
+  ProviderPropType,
+  UrlTile,
+} from 'react-native-maps';
+
+const { width, height } = Dimensions.get('window');
+
+const ASPECT_RATIO = width / height;
+const LATITUDE = 37.78825;
+const LONGITUDE = -122.4324;
+const LATITUDE_DELTA = 0.0922;
+const LONGITUDE_DELTA = LATITUDE_DELTA * ASPECT_RATIO;
+
+class CustomTiles extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      region: {
+        latitude: LATITUDE,
+        longitude: LONGITUDE,
+        latitudeDelta: LATITUDE_DELTA,
+        longitudeDelta: LONGITUDE_DELTA,
+      },
+    };
+  }
+
+  get mapType() {
+    // MapKit does not support 'none' as a base map
+    return this.props.provider === PROVIDER_DEFAULT
+      ? MAP_TYPES.STANDARD
+      : MAP_TYPES.NONE;
+  }
+
+  render() {
+    const { region } = this.state;
+    return (
+      <View style={styles.container}>
+        <MapView
+          provider={this.props.provider}
+          mapType={this.mapType}
+          style={styles.map}
+          initialRegion={region}
+        >
+          <UrlTile
+            urlTemplate="http://c.tile.stamen.com/watercolor/{z}/{x}/{y}.jpg"
+            zIndex={-1}
+						// Empty tileCachePath = defaults to Documents-directory on iOS, 
+						// on Android empty string does not activate tileCachePath at all
+						tileCachePath={''}
```

Comment:
this doesn't look correct. by reading the method `getTileFilename()`, I can see it will return a non-null filename if `tileCachePath` is set to empty string. Also the different behavior between iOS and Android to me seems surprising to me. The behavior should be consistent across platforms I assume

Reply:
@felipecsl I agree ideal would be to have consistency across platforms. Even if in RNM it does not seem be the standard, even for features were underlying platform support is similar e.g. tileSize has not been supported for Android even if it exists in platform the same way as in iOS (this PR fixes that).

However my intent for supporting empty string for tileCachePath was to have native code use a default directory (app documents): this is now implemented for iOS, however for Android empty strings in properties do not seem to get to the URLTile at all, therefore caching is not enabled and the problem that you refer to in `getTileFilename()` does not actually happen. I'm new to RN, RNM and native side coding on both platform - so I was hoping one of the reviewers could guide in how to solve this empty string thing on Android so that I can then add some code to set the default directory correctly.